### PR TITLE
Add file permission validation

### PR DESCRIPTION
**Pull request by Carlgo11**

### CODE REVIEW

File: tests/validate-fs.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# $2, $3, ... = permitted file permissions
+function checkPerm()
+{
+  dir="$1"
+  shift
+  violation=$(find $dir -type f -exec stat -c '%a' '{}' +| grep -vq "$(echo $* | sed 's/ /|/g')")
+  for file in $violation; do
+    echo "Directory '$dir' may only have files with the following permissions: $*"
+    status=1
+  done
```
Comment (phallobst) [inline tests/validate-fs.sh]:
```suggestion
  unset pattern
  for p; do
    pattern+="${pattern+ -a }! -perm $p"
  done
  if find $dir -type f $pattern | grep -q .; then
    echo "Directory '$dir' may only have files with the following permissions: $*"
    status=1
  fi
```
The `-perm` option for `find` is much faster. Also, the error message should only be displayed once per directory, not once for each offending file.

**Review by phallobst — CHANGES_REQUESTED** [review]

File: tests/validate-fs.sh (hunk 1 above)
Comment (Carlgo11) [reply tests/validate-fs.sh]: Thanks! Great to have a shell wizard on the team :smile:

**Comment by phallobst** [comment]
Do we know which versions of `find` and `grep` are in use? I hope it's not some variant of `busybox` or something...

Locally, this script worked fine.

**Comment by Carlgo11** [comment]
Haha yup it's busybox :stuck_out_tongue_winking_eye:

**Comment by Carlgo11** [comment]
The worker should be run on Ubuntu 22.04 LTS so that would make it GNU?

**Review by phallobst — APPROVED** [review]

**Pull request merged**